### PR TITLE
feat: Added wishlist and cart management functionality

### PR DESCRIPTION
**Pull request by mnath30**
Added the following features:
1. Added the wishlist and cart UI
2. Added the wishlist and cart management functionality
3. Added the default pages when the cart and wishlist are empty

Files to be reviewed:
Components
1. CartCard
2. EmptyPage
3. ProductCard
4. TotalPrice
5. WishlistCard

Context
1. CartWishlistContext

Helper functions
1. addToCart
3. addToWishlist
4. cartWishlistInitialState
5. changeQuantity
7. moveToCart
8. moveToWishlist
9. removeFromCart
10. removeFromWishlist

Pages
1. Cart
2. Wishlist

Reducer
1. cartWishlistReducer

### CODE REVIEW

**Review by nerdyspook — APPROVED** [review]
Clean code, liked how you structured your helper functions.

File: src/Components/CartCard/CartCard.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  return (
+    <div className="card cart-card card-horiz">
+      <div className="card-img-horiz">
+        <img className="responsive-img" src={source} alt={information} />
+      </div>
+      <div className="card-body">
+        <h5 className="card-header">{brand}</h5>
+        <p className="card-para">
+          {itemname}
+          <br />
```
Comment (Prabhav-Sharma) [inline src/Components/CartCard/CartCard.jsx]: Suggestion: Not ideal to use `<br>` tags

File: src/Components/EmptyPage/EmptyPage.jsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import "./empty-page.css"
+
+const EmptyPage=({item})=>{
+  const {source,information,message} = item
+  return (
+    <div className="flex-column txt-center">
+      <img className="empty-page-img" src={source} alt={information}/>
+      <h2 className="empty-page-heading">{message}</h2>
```
Comment (Prabhav-Sharma) [inline src/Components/EmptyPage/EmptyPage.jsx:8]: Great idea to make a separate component for empty states

File: src/helper/moveToWishlist.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const moveToWishlist=(item,wishlistArray,cartArray)=>{
+  let newCart,newWishlist;
+  if(wishlistArray.some(element=>element._id===item._id)){
+    newWishlist=[...wishlistArray]
+  }
+  else {
+    newWishlist=[...wishlistArray,{...item,isAddedToWishlist:true,isAddedToCart:false}]
+  }
+  newCart=[...cartArray.filter(element=>element._id!==item._id)]
+  //totalPrice=totalPrice-item.price
```
Comment (Prabhav-Sharma) [inline src/helper/moveToWishlist.js]: Can remove comments if not needed anymore

**Review by Prabhav-Sharma — APPROVED** [review]

File: src/helper/moveToWishlist.js (hunk 3 above)
Comment (prathmeshjagtap) [inline src/helper/moveToWishlist.js]: you can remove commented code if not needed

**Review by prathmeshjagtap — APPROVED** [review]
The code is well structured and very clean, I liked that you have made separate functions to perform cart and wishlist actions such as add to cart, remove cart, also you are storing cart and wishlist data in localstorage I liked that.

**Review by psychoactiv — APPROVED** [review]
Code looks clean
Naming conventions are followed
You can merge this PR

**Pull request merged**